tests/unit/test_params.py

```diff
@@ -20,3 +20,15 @@ def test_params_order(tmp_dir, dvc, dummy_stage):
 
     # params are sorted during dumping, therefore p1 is first
     assert list(dvc.params.show()[""]) == ["p1.yaml", p2_path, "params.yaml"]
+
+
+def test_repro_unicode(tmp_dir, dvc, dummy_stage):
+    tmp_dir.gen({"settings.json": '{"Ω_value": 1}'})
+    dummy_stage(params=[{"settings.json": ["Ω_value"]}])
+    (stage,) = dvc.reproduce(dry=True)
+    stage.cmd = "foo"
+    stage.dvcfile._lockfile.dump(stage)
```

Comment:
Let's just use `stage.dump()` to both files. This way, both the `remove_stage` will be used. 🙂

Reply:
I thought `dummy_stage()` already did it

Reply:
*it doesn't create a `dvc.lock` file but rather only dumps the stage itself to the `dvc.yaml` 
https://github.com/iterative/dvc/blob/08300d3bcf51f7bd9c8509924110416461b11b60/tests/conftest.py#L148-L151

Reply:
Okay, let's leave it then. I made a mistake on that `stage.dump` not dumping lockfile by default.  Considering the callee of it, it's not a bug, but we should keep the behavior same here. I'll look into later.